filament/backend/src/vulkan/VulkanDriver.cpp

```diff
@@ -325,6 +325,10 @@ ShaderModel VulkanDriver::getShaderModel() const noexcept {
 }
 
 void VulkanDriver::terminate() {
+    // Flush and wait here to make sure all queued commands are executed and resources that are tied
+    // to those commands are no longer referenced.
+    finish(0);
```

Comment:
is there any risk this could block forever?

Reply:
This is essentially the same as calling `flushAndWait()` right before engine teardown, so there is a risk, but not any more risky than a pattern that is actively being used.